crates/uv/src/commands/project/lock.rs

```diff
@@ -877,6 +878,42 @@ impl ValidatedLock {
             return Ok(Self::Versions(lock));
         }
 
+        // Catch a lockfile where the union of fork markers doesn't cover the supported
+        // environments.
+        //
+        // We subset by requires-python, since the space outside of it does not matter. The fork
+        // markers however should contain the requires-python value, if they don't it wouldn't be
+        // a problem on sync, but it indicates a broken lockfile where we should recompute the
+        // fork-markers (the versions are preserved, so the churn is low).
+        let fork_markers_union = if lock.fork_markers().is_empty() {
+            requires_python.to_marker_tree()
+        } else {
+            let mut fork_markers_union = MarkerTree::FALSE;
+            for fork_marker in lock.fork_markers() {
+                fork_markers_union.or(fork_marker.pep508());
+            }
+            fork_markers_union
+        };
+        let mut environments_union = if let Some(environments) = environments {
+            let mut environments_union = MarkerTree::FALSE;
+            for fork_marker in environments.as_markers() {
+                environments_union.or(*fork_marker);
+            }
+            environments_union
+        } else {
+            MarkerTree::TRUE
+        };
+        // We respect requires-python in addition to the environments the use specified.
+        environments_union.and(requires_python.to_marker_tree());
+        if fork_markers_union != environments_union {
```

Comment:
Do they have to be exactly equal...? Could one be a superset?

Reply:
I'm on the fence whether to do an equality checks or a contains check, i like the equality check because it should catch more bad markers, but we can already revert that to `if fork_markers_union.negate().is_disjoint(environments_union)`.

Reply:
Lets do the more conservative thing for now... I'm just worried that this could have a significant impact and doesn't have a ton of testing, does it?

Reply:
I would prefer to change this but don't feel confident enough doing it myself, so it may miss this release.

Reply:
done